packages/storage/src/types/upload.ts

```diff
@@ -22,15 +35,36 @@ export type UploadProgressEvent = {
 };
 
 export type IpfsUploaderOptions = {
+  /**
+   * Mapping of URL schemes to gateway URLs to resolve to
+   */
   gatewayUrls?: GatewayUrls;
+  /**
+   * Whether or not to replace any URLs with schemes with resolved URLs before upload
+   */
   uploadWithGatewayUrl?: boolean;
 };
 
 export type IpfsUploadBatchOptions = {
+  /**
+   * If specified, will rewrite file names to numbers for use on-chain
```

Comment:
could be more explicit here - like "useful to use with NFT contracts that map token ids to files"